File: src/1. Two Sum/main.ts
// Solved by @kitanoyoru
// https://leetcode.com/problems/two-sum/

const twoSum = (nums: number[], target: number): number[] => {
  for (let i = 0; i < nums.length; ++i) {
    for (let j = 0; j < nums.length; ++j) {
      if (i != j) {
        if (nums[i] + nums[j] == target) {
          return [i, j]
        }
      }
    }
  }
}

File: src/1. Two Sum/main.ex
defmodule Solution do
  @spec two_sum(nums :: [integer], target :: integer) :: [integer]
  def two_sum(nums, target) do
    Enum.reduce_while(nums, {%{}, 0}, fn n, {map, i} ->
        complement = Map.get(map, target - n)
        if complement do
            {:halt, [i, complement]}
        else
            {:cont, {Map.put(map, n, i), i + 1}}
        end
    end)
  end
end
